src/zenml/entrypoints/entrypoint.py

```diff
@@ -39,12 +40,23 @@ def main() -> None:
     # is not wrapped in a function or an `if __name__== "__main__":` check)
     constants.SHOULD_PREVENT_PIPELINE_EXECUTION = True
 
+    source_utils.set_custom_source_root(source_root="custom_source_root")
+
     # Read the source for the entrypoint configuration class from the command
     # line arguments
     parser = argparse.ArgumentParser()
     parser.add_argument(f"--{ENTRYPOINT_CONFIG_SOURCE_OPTION}", required=True)
+    parser.add_argument("--env", default="")
     args, remaining_args = parser.parse_known_args()
 
+    # Set environment variables from the --env parameter
+    env_vars = args.env.split(",")
```

Comment:
Not a fan of this at all. If anything, this should be in a subclass of the `StepEntrypointConfiguration` for databricks, not hacked into the general code.